presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +263,18 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
+            if (millis >= GREGORIAN_CALENDAR_INTRODUCTION) {
+                return new Date(millis);
+            }
+            // the chronology used by default by Joda is not appropriate for dates preceding the introduction of the Gregorian calendar,
+            // so for such cases we are falling back to the more expensive GregorianCalendar; note that Joda also has a chronology that
+            // works for older dates, but it uses a slightly different algorithm, so we are sticking with the standard library
```

Comment:
What are the end user visible symptoms of these differences? are they test covered?